docs/source/data/errors.mdx

```diff
@@ -604,21 +604,60 @@ In this case, the error above is reported to Apollo Studio as:
+- If Apollo Server cannot parse the request into a legal GraphQL document and validate it against your schema, use status code 400. This also includes other problems with the request, such as attempting to send a batched HTTP request when `allowBatchedHttpRequests` is not enabled, or if CSRF prevention blocks the request.
+- If an invalid HTTP method is used (`GET` with a mutation, or any HTTP method other than `GET` or `POST`), use status code 405.
+- If your `context` function throws, use status code 500.
+- If there is an unexpected error during the processing of the request (either a bug in Apollo Server, or a plugin hook that throws), use status code 500.
+- Otherwise, use status code 200. This is essentially the case where the server is able to execute the GraphQL operation, and execution completes successfully (possibly with some resolver-specific errors).
 
-As an example, here is how you could set a custom response header and status code based on a GraphQL error:
+You can affect the HTTP status code or set response headers either by throwing an error in a resolver or `context` function, or by writing a [plugin](/apollo-server/integrations/plugins). Be aware that GraphQL client libraries may not treat all response status codes the same, and so it will be up to your team to decide what patterns to use. We strongly encourage you to avoid setting the HTTP status code based on an error thrown from a resolver; setting the status code from context creation errors can be more useful.
+
+To affect the HTTP status code and response headers from an error throw in a resolver or context function, throw a `GraphQLError` with an `http` extension:
+
+<MultiCodeBlock>
+
+```ts
+import { GraphQLError } from 'graphql';
+
+const resolvers = {
+  Query: {
+    someField() {
+      throw new GraphQLError('the error message', {
+        extensions: {
+          code: 'SOMETHING_BAD_HAPPENED',
+          http: {
+            status: 404,
+            headers: new Map([
+              ['some-header', 'it was bad'],
+              ['another-header', 'seriously'],
+            ]),
+          },
+        },
+      });
+    }
+  }
+}
+```
+
+</MultiCodeBlock>
+
+You do not need to include `status` unless you want to override the default (200 for a resolver, 500 for a `context` function). You do not need to include `headers` unless you want to set headers. You should not rely on the behavior of what happens if multiple resolvers throw errors which set the status or set the same header; if you need to handle this case, write a plugin instead. (However, different resolver errors can set different headers.) Headers should be provided in a `Map` with lowercase names.
```

Comment:
```suggestion
You don't need to include `status` unless you want to override the default status code (200 for a resolver or 500 for a `context` function).   The optional `headers` field should provide a `Map` with lowercase header names.

If your setup includes multiple resolvers throwing errors that set the same status code or header, we recommend writing a plugin instead (as shown below).
```

Reply:
I would like to be explicit that this behavior is 'undefined' in the sense that we could change it in a release without it counting as backwards-incompatible.

Reply:
How does something like this feel instead (being more clear this isn't a consistent thing):
```suggestion
You don't need to include `status` unless you want to override the default status code (200 for a resolver or 500 for a `context` function). The optional `headers` field should provide a `Map` with lowercase header names.

If your setup includes multiple resolvers which throw errors that set status codes or set the same header, the errors thrown are not guaranteed to be consistent. Instead, we recommend writing a plugin (as shown below).
```

Reply:
How about 

```suggestion
You don't need to include `status` unless you want to override the default status code (200 for a resolver or 500 for a `context` function). The optional `headers` field should provide a `Map` with lowercase header names.

If your setup includes multiple resolvers which throw errors that set status codes or set the same header, Apollo Server's may resolve this conflict in an arbitrary way which may change in future versions. Instead, we recommend writing a plugin (as shown below).
```

Reply:
Small suggested tweaks for styling, but otherwise feels great:
```suggestion
You don't need to include `status` unless you want to override the default status code (200 for a resolver or 500 for a `context` function). The optional `headers` field should provide a `Map` with lowercase header names.

If your setup includes multiple resolvers which throw errors that set status codes or set the same header, Apollo Server might resolve this conflict in an arbitrary way (which could change in future versions). Instead, we recommend writing a plugin (as shown below).
```